Miscellaneous Updates

simonwiles opened the pull request:
This is a couple of minor linting-type fixes, and a perhaps controversial one (efe081c1c94a931d19e5b9b1958e1f09588deb3b).  There was supposed to be more, but this has been languishing for a week and I've not added anything else to it yet, so I'm PRing now, and will do another PR later.

simonwiles committed:
Better header background

The first time I saw this header I wanted to fix this.  Imo this looks tonnes better, but 1) you may disagree? and 2) it isn't exactly the same as the spotlight header (although that ship is already on its way out of port, to some degree).  Either of these considerations may be sufficient to nix this -- it's not very important, it just bugged me a little.

simonwiles committed:
Fix a linting error

`JSON.parse` is quite happy to accept `null` as an argument in practice, but it's not supposed to in theory 🤷.

simonwiles committed:
Fix a linting error

`JSON.parse` is quite happy to accept `null` as an argument in practice, but it's not supposed to in theory 🤷.

simonwiles committed:
Remove preloaded image

This was only for the splash screen component that I see was removed in November 2023... :)

simonwiles committed:
Remove preloaded image

This was only for the splash screen component that I see was removed in November 2023... :)

simonwiles committed:
Layout for documentation pages should extend base layout

My attention was initially drawn to this because the head element in this layout template was missing the `prefix="dcterms: http://purl.org/dc/terms/#"` attribute, so the `<meta name="dcterms.modified" ...>` element was invalid.  But it was also missing all the `apple-touch-icon`s and such.

The only problem with this is that the base layout includes the script snippet that reads the preferred theme from localStorage, but that's better dealt with another way.

simonwiles committed:
Layout for documentation pages should extend base layout

My attention was initially drawn to this because the head element in this layout template was missing the `prefix="dcterms: http://purl.org/dc/terms/#"` attribute, so the `<meta name="dcterms.modified" ...>` element was invalid.  But it was also missing all the `apple-touch-icon`s and such.

The only problem with this is that the base layout includes the script snippet that reads the preferred theme from localStorage, but that's better dealt with another way.

simonwiles committed:
Allow passing `theme` to base layout

This allows pages to force a specified theme.  I still think the theme selector should probably be removed, and I still don't like the use of the themes to differentiate the listen and perform modes, but I know I'm not winning either of those arguments :)

simonwiles committed:
Allow passing `theme` to base layout

This allows pages to force a specified theme.  I still think the theme selector should probably be removed, and I still don't like the use of the themes to differentiate the listen and perform modes, but I know I'm not winning either of those arguments :)

broadwell reviewed (changes requested):
You're right, everything looks great except perhaps https://github.com/sul-cidr/pianolatron/commit/efe081c1c94a931d19e5b9b1958e1f09588deb3b -- which is functionally an improvement, for sure, but it does kind of go against what I see as the main priority for the header behavior, which is just to minimize the degree of visual surprise when going back and forth between the SUPRA exhibit and the Pianolatron search and play modes (obviously the perform mode is kind of the odd duck in this regard).
Choosing either path at this point involves a lot of speculation about stakeholder preferences and the trajectory of the Spotlight integration, but I think I'd prefer to leave https://github.com/sul-cidr/pianolatron/commit/efe081c1c94a931d19e5b9b1958e1f09588deb3b out for now.
Also it's odd that the Netlify preview didn't build, but I can't see how that could be anything but a problem on their end.

simonwiles commented:
Okay, no problem, offending commit has been dropped 👍️.

broadwell committed:
Merge branch 'misc' of https://github.com/sul-cidr/pianolatron into misc

broadwell committed:
Quickfix for 88-note roll zk199jz1564

Previously, this roll was one of the gen1 88-note rolls (actually it was the only one) with a strong-armed shift in tracker alignment/MIDI assignment hard-coded into the roll-wrangler script to make the sustain pedal holes be interpreted properly. Due to the recent improvements to the roll image parser code, this is no longer necessary!
